### ISSUE TITLE
Fuel level calculation based on distance and other

### ISSUE THREAD

**Issue by Abyss777** [tananaev/traccar#3336]
I liked the idea from https://www.traccar.org/forums/topic/help-with-computed-attributes/
I think that calculated "fuel level"/"spent fuel" based on distance and theoretical consumption might be useful and rather general and might be implemented in code.

There are two options for me:
- Use fake very large tank and subtract calculated spent fuel for every position and save it like `fuelLevel`
Or make tank real and implement manual refuel. 
- Save some not accumulated "spent fuel" attribute in every position (like distance) and summarize it in reports.

**Comment by tananaev** [comment]
I think we need to implement it properly. If device reports consumption and not fuel level, we should use that in reports. That's because some devices can report it natively.

**Comment by Abyss777** [comment]
You are right, in case device reports average consumption we simply can multiply traveled distance on average consumption, but I think usually it reports instant consumption and we should go through all positions and accumulate it.

And as final fallback we can simply multiply traveled distance on average consumption if it is set in device attributes.

**Comment by tananaev** [comment]
Agree. I think there are some devices that do report average consumption.

**Comment by LucasBSC** [comment]
Is this feature somehow already implemented?
